syft/rekor/queryRekor/utils.go

```diff
@@ -0,0 +1,274 @@
+	if err != nil {
+		log.Debug("Error unmarshaling json entry body to intoto")
+		return nil, err
+	}
+
+	// TODO: validate
+	if *intoto.APIVersion != "0.0.1" {
+		log.Debug("Error parsing rekor entry body")
+		err := fmt.Sprintf("intoto schema version %v not supported", *intoto.APIVersion)
+		return nil, errors.New(err)
+	}
+
+	specBytes, err := json.Marshal(intoto.Spec)
+	if err != nil {
+		log.Debug("error marshaling intoto spec to json")
+		return nil, err
+	}
+
+	intotoV001 := new(models.IntotoV001Schema)
+	err = intotoV001.UnmarshalBinary(specBytes)
+	if err != nil {
+		log.Debug("Error unmarshaling intoto spec to intotoV001 schema")
+		return nil, err
+	}
+
+	return intotoV001, err
+}
+
+// parse the attestation and validate that it fits the inTotoAttestation type
+func parseAndValidateAttestations(entries []models.LogEntryAnon) ([]inTotoAttestation, error) {
+
+	var atts []inTotoAttestation
+	for _, entry := range entries {
+		attAnon := entry.Attestation
+		if attAnon == nil {
+			log.Debug("Error parsing attestation")
+			return nil, errors.New("log entry attestation is nil")
+		}
+
+		// this decodes the base64 string
+		attDecoded := string(attAnon.Data)
+
+		att := new(inTotoAttestation)
+		err := json.Unmarshal([]byte(attDecoded), att)
+		if err != nil {
+			log.Debug("Error parsing attestation")
+			return nil, err
+		}
+
+		// validation the entry is an sbom entry is done by validating attestation struct
+		if att.PredicateType == "" {
+			log.Debugf("Rekor entry %v could not be parsed as an sbom entry", *entry.LogIndex)
+			continue
+		}
+		if att.PredicateType != "http://lumjjb/sbom-documents" {
+			log.Debugf("Rekor entry %v could not be parsed as an sbom entry", *entry.LogIndex)
+			continue
+		}
+		if att.Predicate.Sboms == nil {
+			log.Debugf("Rekor entry %v could not be parsed as an sbom entry", *entry.LogIndex)
+			continue
+		}
+
+		atts = append(atts, *att)
+	}
+
+	return atts, nil
+}
+
+func parseSboms(spdxBytesList map[*inTotoAttestation][]byte) (map[*inTotoAttestation]spdx.Document2_2, error) {
+
+	sboms := make(map[*inTotoAttestation]spdx.Document2_2)
+	for att, spdxBytes := range spdxBytesList {
+		//remove all SHA512 hashes because spdx/tools-golang does not support
+		// PR fix is filed but not merged
+
+		regex, err := regexp.Compile("\n.*SHA512.*")
+		if err != nil {
+			log.Debug("Error decoding sbom to syft type")
+			return nil, err
+		}
+
+		spdxBytes = regex.ReplaceAll(spdxBytes, nil)
+		sbom, err := tvloader.Load2_2(bytes.NewReader(spdxBytes))
+		if err != nil {
+			log.Debug("Error loading sbom bytes into spdx type")
+			return nil, err
+		}
+
+		sboms[att] = *sbom
+	}
+
+	return sboms, nil
+}
+
+func formatSbomPackageName(att *inTotoAttestation) string {
+	repo := att.Predicate.BuildMetadata.ArtifactSourceRepo
+	commit := att.Predicate.BuildMetadata.ArtifactSourceRepoCommit
+	if repo == "" {
+		return ""
+	}
+	if commit == "" {
+		return repo
+	}
+	return fmt.Sprintf("%v@%v", repo, commit)
+}
```

Comment:
this isn't quite always going to be accurate but not don't think there's a way to do this in a mathematically defined manner right now, going to be heuristic based - we can talk a bit more about this.